utility/environment_check.py

```diff
@@ -44,46 +58,32 @@ def environment_checker(request):
     get_status_before_execution()
 
 
+def assign_get_values(env_status_dict, key, kind):
+    env_status_dict[key] = kind.get(all_namespaces=True)['items']
+
+
+def get_environment_status(env_dict):
+    """
+
+    """
+    with ThreadPoolExecutor(max_workers=len(KINDS)) as executor:
+        for key, kind in zip(env_dict.keys(), KINDS):
+            executor.submit(assign_get_values, env_dict, key, kind)
+
+
 def get_status_before_execution():
     """
 
     """
-    ENV_STATUS_PRE['pod'] = POD.get(all_namespaces=True)['items']
-    ENV_STATUS_PRE['sc'] = SC.get(all_namespaces=True)['items']
-    ENV_STATUS_PRE['cephfs'] = CEPHFILESYSTEM.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_PRE['cephbp'] = CEPHBLOCKPOOL.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_PRE['pv'] = PV.get(all_namespaces=True)['items']
-    ENV_STATUS_PRE['pvc'] = PVC.get(all_namespaces=True)['items']
-    ENV_STATUS_PRE['secret'] = SECRET.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_PRE['namespace'] = NS.get(all_namespaces=True)['items']
+    get_environment_status(ENV_STATUS_PRE)
 
 
 def get_status_after_execution():
     """
 
     """
-    ENV_STATUS_POST['pod'] = POD.get(all_namespaces=True)['items']
-    ENV_STATUS_POST['sc'] = SC.get(all_namespaces=True)['items']
-    ENV_STATUS_POST['cephfs'] = CEPHFILESYSTEM.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_POST['cephbp'] = CEPHBLOCKPOOL.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_POST['pv'] = PV.get(all_namespaces=True)['items']
-    ENV_STATUS_POST['pvc'] = PVC.get(all_namespaces=True)['items']
-    ENV_STATUS_POST['secret'] = SECRET.get(
-        all_namespaces=True
-    )['items']
-    ENV_STATUS_POST['namespace'] = NS.get(
-        all_namespaces=True
-    )['items']
+    get_environment_status(ENV_STATUS_POST)
+
     pod_diff = DeepDiff(
```

Comment:
what about the other resources?

Reply:
what other resources?
Be more descriptive, please

Reply:
sorry, I meant what about getting the diff of the other resources - pv, pvc, secret, etc

Reply:
oh sorry again, it's down below :)